### PR TITLE
Add info for the `flutter_reactive_widget` state management library

### PR DESCRIPTION
**Pull request by lukehutch**
_Description of what this PR is changing or adding, and why:_

Added info for `flutter_reactive_widget`, my state management library

_Issues fixed by this PR (if any):_

N/A

## Presubmit checklist
- [X] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [X] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [X] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/development/data-and-backend/state-mgmt/options.md [hunk 1]
```diff
@@ -288,3 +288,19 @@ A simple but powerful state management solution inspired by SolidJS.
 [Official Documentation]: https://docs.page/nank1ro/solidart
 [solidart package]: {{site.pub-pkg}}/solidart
 [flutter_solidart package]: {{site.pub-pkg}}/flutter_solidart
+
+## flutter_reactive_widget
+
+An ultra-low-boilerplate solution for state management, which defines
```
Comment (sfshaza2) [inline src/development/data-and-backend/state-mgmt/options.md:294]:
```suggestion
An ultra-low-boilerplate solution for state management,
flutter_reactive_widget defines
```

File: src/development/data-and-backend/state-mgmt/options.md [hunk 2]
```diff
@@ -288,3 +288,19 @@ A simple but powerful state management solution inspired by SolidJS.
...
+`ReactiveWidget` and `ReactiveValue`. Any read of a `ReactiveValue`'s
+value within a `ReactiveWidget` definition automatically causes the
+`ReactiveWidget` to listen for changes on the `ReactiveValue`.
+
+Also includes a definition for `PersistentReactiveValue`, a subclass
+of `ReactiveValue` whose latest value is persisted, surviving app
+restarts.
```
Comment (sfshaza2) [inline src/development/data-and-backend/state-mgmt/options.md:301]:
```suggestion
of `ReactiveValue` whose latest value persists, surviving app
restarts.
```

**Review by sfshaza2 — APPROVED** [review]
lgtm, after a couple minor tweaks

**Comment by lukehutch** [comment]
Thanks!

**Pull request merged**